### ISSUE TITLE
conan export-pkg touches remotes in 1.7.0

### ISSUE THREAD

**Issue by brianembry** [conan-io/conan#3432]
1.6.0 doesn't appear to have this behavior, but if you call "conan export-pkg" with a build_requires and a package override? it appears that conan will make a connection to available remotes.  Here's the TB where I added an assert in _call_remote()

This "broke" us because the box couldn't connect to bintray.  Not a huge deal (as I can just remove the remote), but I wouldn't expect a remote to be touched with export-pkg if the package is already installed.

```
$ conan export-pkg . brian/test
Exporting package recipe
blah/0.7@brian/test: The stored package has not changed
Traceback (most recent call last):
  File "/home/brian/.local/lib/python2.7/site-packages/conans/client/command.py", line 1251, in run
    method(args[0][1:])
  File "/home/brian/.local/lib/python2.7/site-packages/conans/client/command.py", line 694, in export_pkg
    channel=channel)
  File "/home/brian/.local/lib/python2.7/site-packages/conans/client/conan_api.py", line 82, in wrapper
    return f(*args, **kwargs)
  File "/home/brian/.local/lib/python2.7/site-packages/conans/client/conan_api.py", line 382, in export_pkg
    profile=profile, force=force)
  File "/home/brian/.local/lib/python2.7/site-packages/conans/client/manager.py", line 48, in export_pkg
    remote_name=None, recorder=self._recorder, workspace=None)
  File "/home/brian/.local/lib/python2.7/site-packages/conans/client/graph/graph_manager.py", line 115, in load_graph
    processed_profile=processed_profile)
  File "/home/brian/.local/lib/python2.7/site-packages/conans/client/graph/graph_manager.py", line 181, in _load_graph
    binaries_analyzer.evaluate_graph(graph, build_mode, update, remote_name)
  File "/home/brian/.local/lib/python2.7/site-packages/conans/client/graph/graph_binaries.py", line 148, in evaluate_graph
    self._evaluate_node(node, build_mode, update, evaluated_references, remote_name)
  File "/home/brian/.local/lib/python2.7/site-packages/conans/client/graph/graph_binaries.py", line 106, in _evaluate_node
    remote_info = self._get_package_info(package_ref, r)
  File "/home/brian/.local/lib/python2.7/site-packages/conans/client/graph/graph_binaries.py", line 24, in _get_package_info
    remote_info = self._remote_manager.get_package_info(package_ref, remote)
  File "/home/brian/.local/lib/python2.7/site-packages/conans/client/remote_manager.py", line 171, in get_package_info
    return self._call_remote(remote, "get_package_info", package_reference)
  File "/home/brian/.local/lib/python2.7/site-packages/conans/client/remote_manager.py", line 276, in _call_remote
    assert(0)
AssertionError
```

**Comment by memsharded** [comment]
Yes, we need to check it. Where are you defining the build_requires, in the recipe itself? And you ran first the ``conan install`` and the build_requires was installed locally? Please give a bit more of details, so we can investigate. If a bug, we will try to release a fix asap, in 1.7.1

**Comment by brianembry** [comment]
Yeah, conan install was run, package was built, then export-pkg was called and a connection is made to the remotes.  Remote was hit if package had a build_requires, which makes sense.

If you're expecting to hit the remote here that's fine.

$ git clone git@github.com:conan-community/conan-openssl.git
$ cd conan-openssl
$ conan install .
$ conan export-pkg . brian/test <--will hit remotes again

**Comment by memsharded** [comment]
Thanks for the feedback, I have identified the issue, and I think it can be improved, will try to submit a fix for 1.7.1
